experimental/Schemes/IdealSheaves.jl

```diff
@@ -1210,3 +1210,105 @@ function saturation(I::IdealSheaf, J::IdealSheaf)
+with restrictions
+  1: ideal(x_2_1, x_3_1)
+  2: ideal(x_2_2, x_3_2)
+  3: ideal(1, x_3_3)
+  4: ideal(x_2_4, 1)
+```
+"""
+function IdealSheaf(X::NormalToricVariety, I::MPolyIdeal)
+  @req base_ring(I) === cox_ring(X) "ideal must live in the cox ring of the variety"
+
+  # We currently only support this provided that the following conditions are met:
+  # 1. All maximal cones are smooth, i.e. the fan is smooth/X is smooth.
+  # 2. The dimension of all maximal cones matches the dimension of the fan.
+  @req is_smooth(X) "Currently, ideal sheaves are only supported for smooth toric varieties"
+  @req all(m -> dim(m) == dim(X), maximal_cones(X)) "Currently, ideal sheaves require that all maximal cones have the dimension of the variety"
+
+  ideal_dict = IdDict{AbsSpec, Ideal}()
+
+  # We need to dehomogenize the ideal I ⊂ S = ℂ[xᵨ| ρ ∈ Σ(1)] to the local 
+  # charts Uₛ, s ⊂ Σ a cone. To this end we use Proposition 5.2.10 from 
+  # Cox-Little-Schenck, p. 223ff. Abstractly, the chart 
+  # Uₛ is isomorphic to ℂˢ⁽¹⁾ = Spec(ℂ[xᵨ | ρ ∈ s(1)]). But these are 
+  # not the coordinates we use for the affine charts here. They 
+  # correspond to the `hilbert_basis(dual_cone(U_s))`.
+
+  # TODO: In the long run we should think about making the creation of the 
+  # following dictionary lazy. But this requires partial rewriting of the 
+  # ideal sheaves as a whole, so we postpone it for the moment.
+
+  for (k, U) in enumerate(affine_charts(X))
+
+    # We first create the morphism Φₛ* from p. 224, l. 3.
+    indices = [j for j in 1:nrays(X) if ray_indices(maximal_cones(X))[k,j]]
+    help_ring, x_rho = polynomial_ring(QQ, ["x_$j" for j in indices])
+    imgs_phi_star = [j in indices ? x_rho[findfirst(k->k==j, indices)] : one(help_ring) for j in 1:nrays(X)]
+    phi_s_star = hom(cox_ring(X), help_ring, imgs_phi_star)
+
+    # Now we need to create the inverse of α*.
+    imgs_alpha_star = elem_type(help_ring)[]
+    for m in hilbert_basis(dual_cone(U))
+      img = one(help_ring)
+      for j in 1:length(indices)
+        u_rho = matrix(ZZ,rays(X))[indices[j],:]
+        expo = ZZ(_my_dot(m, u_rho))
+        img = img * x_rho[j]^expo
+      end
+      push!(imgs_alpha_star, img)
+    end
+    alpha_star = hom(OO(U), help_ring, imgs_alpha_star)
+
+    # TODO: There should be better ways to create this map!
+    # Presumably, one can invert the matrix with the `expo`s as entries?
+    # @larskastner If you can confirm this, let's change this.
+    alpha_star_inv = inverse(alpha_star)
+    ideal_dict[U] = ideal(OO(U), [alpha_star_inv(phi_s_star(g)) for g in gens(I)])
+  end
+
+  return IdealSheaf(X, ideal_dict, check=true) # TODO: Set to false eventually.
+end
+
+function _my_dot(v::PointVector{ZZRingElem}, u::ZZMatrix)
+  n = length(v)
+  @req n == length(u) "matrix sizes do not match"
+  return sum(v[i]*u[1,i] for i in 1:n; init=zero(QQ))
+end
+
+function _generic_blow_up(v::Any, I::Any)
+  error("Not yet supported")
+end
```

Comment:
The code on ideal sheaves is still experimental, while the `blow_up` function in the toric varieties is in `src`.

I understand that there was a rule "`src` must never call `experimental`" or similar. I discussed this with @fingolfin yesterday, and he proposed this work-around. It ensure that even if you disabled all `experimental` code, the toric varieties code would still run except for raising the error "Not yet supported" exactly where the call into `experimental` would have happened.

Reply:
I don't understand. If this wasn't here you would also get an error, right? Also this function is in `experimental`, so it would be disabled too if `experimental` is gone?

Reply:
I also noticed this and adjusted it. Let me push the changes that I have by now. One sec.

Reply:
If I did it correctly, then this should now be in `standard_constructions.jl`, right below the `blow_up` function.